aospy/utils/times.py

```diff
@@ -165,7 +167,9 @@ def numpy_datetime_range_workaround(date, min_year):
         permissible dates, otherwise a datetime.datetime object with the year
         offset to the earliest allowable year.
     """
-    if date < pd.Timestamp.min:
+    min_yr_in_range = pd.Timestamp.min.year < min_year < pd.Timestamp.max.year
+    max_yr_in_range = pd.Timestamp.min.year < max_year < pd.Timestamp.max.year
+    if not (min_yr_in_range and max_yr_in_range):
```

Comment:
Just confirming I'm understanding the situation correctly: if any part of the loaded data is outside of the valid range, the decoding will be unable to use the standard numpy datetimes.  In that case, we apply our offset so that it "starts" near the beginning of the valid date range.  

Importantly, the loaded data includes the whole range of data loaded onto disk, even if the date range we're interested in is a subset of that, and even if that subset is itself entirely within the valid range.

If that's correct then I think your logic is correct.